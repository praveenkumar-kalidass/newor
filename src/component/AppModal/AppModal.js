import React from 'react';
import { Modal } from 'react-native';
import PropTypes from 'prop-types';

import { ModalContainer, ModalOverlay } from './AppModal.style';

const AppModal = ({
  visible,
  children,
}) => (
  <Modal
    animationType="slide"
    transparent
    visible={visible}
  >
    <ModalOverlay>
      <ModalContainer>
        {children}
      </ModalContainer>
    </ModalOverlay>
  </Modal>
);

AppModal.propTypes = {
  visible: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
};

export default AppModal;
