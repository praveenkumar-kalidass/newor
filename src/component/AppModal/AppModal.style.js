import styled from 'styled-components/native';
import { Box, Flex, IconButton } from 'native-base';

export const ModalOverlay = styled(Flex).attrs(({ theme }) => ({
  flex: 1,
  direction: 'row',
  p: '5',
  justifyContent: 'center',
  bg: `${theme.color.BACKGROUND_0}:alpha.50`,
}))``;

export const ModalContainer = styled(Box).attrs(({ theme }) => ({
  borderRadius: 20,
  flex: 1,
  alignSelf: 'center',
  p: 5,
  bg: theme.color.MODAL,
  elevation: 1,
  shadowColor: theme.color.BACKGROUND_0,
}))``;

export const CloseButton = styled(IconButton).attrs(({ theme }) => ({
  variant: 'ghost',
  _icon: {
    color: theme.color.SECONDARY_100,
  },
  _pressed: {
    bg: `${theme.color.SECONDARY_100}:alpha.10`,
  },
}))``;
