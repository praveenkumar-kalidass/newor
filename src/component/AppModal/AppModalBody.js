import React from 'react';
import PropTypes from 'prop-types';
import { Flex } from 'native-base';

const AppModalBody = ({ children }) => (
  <Flex justify="center" align="center">
    {children}
  </Flex>
);

AppModalBody.propTypes = {
  children: PropTypes.node.isRequired,
};

export default AppModalBody;
