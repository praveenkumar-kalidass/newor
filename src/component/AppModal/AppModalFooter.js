import React from 'react';
import PropTypes from 'prop-types';
import { Divider, Flex } from 'native-base';

const AppModalFooter = ({ children }) => (
  <>
    <Divider my="5" />
    <Flex direction="row" justify="flex-end" align="center">
      {children}
    </Flex>
  </>
);

AppModalFooter.propTypes = {
  children: PropTypes.node.isRequired,
};

export default AppModalFooter;
