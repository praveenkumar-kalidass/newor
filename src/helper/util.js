import CONSTANT from 'constant';

export const formatCurrency = (value) => {
  const formatter = new Intl.NumberFormat('hi-IN', { style: 'currency', currency: 'INR' });
  return formatter.format(value);
};

export const unformatCurrency = (value) => {
  const withoutSymbol = value.replace(CONSTANT.APP_LITERAL.RUPEE_SYMBOL, '');
  const currency = withoutSymbol.replace(/,/g, '');
  return parseFloat(currency);
};

export const formatUrlParams = (data) => Object.keys(data).map((key) => (
  `${encodeURIComponent(key)}=${encodeURIComponent(data[key])}`
)).join('&');
