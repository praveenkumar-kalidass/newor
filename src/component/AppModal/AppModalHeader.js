import React from 'react';
import PropTypes from 'prop-types';
import { Flex, CloseIcon, Divider } from 'native-base';

import { CloseButton } from './AppModal.style';

const AppModalHeader = ({ children, showClose, onClose }) => (
  <>
    <Flex direction="row" justify="space-between" align="center">
      {children}
      <If condition={showClose}>
        <CloseButton icon={<CloseIcon />} onPress={onClose} />
      </If>
    </Flex>
    <Divider my="2" />
  </>
);

AppModalHeader.propTypes = {
  children: PropTypes.node.isRequired,
  showClose: PropTypes.bool,
  onClose: PropTypes.func,
};

AppModalHeader.defaultProps = {
  showClose: true,
  onClose: () => {},
};

export default AppModalHeader;
