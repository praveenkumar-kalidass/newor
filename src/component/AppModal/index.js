import AppModal from './AppModal';
import AppModalBody from './AppModalBody';
import AppModalFooter from './AppModalFooter';
import AppModalHeader from './AppModalHeader';

AppModal.Body = AppModalBody;
AppModal.Footer = AppModalFooter;
AppModal.Header = AppModalHeader;

export default AppModal;
