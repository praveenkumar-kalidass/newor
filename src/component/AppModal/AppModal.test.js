import { Text } from 'native-base';
import React from 'react';

import { render } from 'test/util';
import AppModal from './index';

describe('AppModal', () => {
  it('should match snapshot', () => {
    const container = render(
      <AppModal visible>
        <AppModal.Header showClose>
          <Text>Header</Text>
        </AppModal.Header>
        <AppModal.Body>
          <Text>Body</Text>
        </AppModal.Body>
        <AppModal.Footer>
          <Text>Footer</Text>
        </AppModal.Footer>
      </AppModal>,
    );

    expect(container).toMatchSnapshot();
  });
});
