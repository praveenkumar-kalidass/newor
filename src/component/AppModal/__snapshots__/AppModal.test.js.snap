// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppModal should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <Modal
        animationType="slide"
        hardwareAccelerated={false}
        transparent={true}
        visible={true}
      >
        <View
          dataSet={Object {}}
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.5)",
                "display": "flex",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 20,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 20,
              },
              Array [
                Object {},
              ],
            ]
          }
        >
          <View
            dataSet={Object {}}
            elevation={1}
            shadowColor="#000000"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "#eeeeee",
                  "borderRadius": 20,
                  "flex": 1,
                  "paddingBottom": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 20,
                },
                Array [
                  Object {},
                ],
              ]
            }
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Header
              </Text>
              <View
                accessibilityRole="button"
                accessible={true}
                collapsable={false}
                colorScheme="primary"
                dataSet={Object {}}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 4,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingBottom": 10,
                      "paddingLeft": 10,
                      "paddingRight": 10,
                      "paddingTop": 10,
                    },
                    Array [
                      Object {},
                    ],
                  ]
                }
              >
                <RNSVGSvgView
                  accessibilityRole="image"
                  align="xMidYMid"
                  bbHeight={20}
                  bbWidth={20}
                  color={4278199178}
                  dataSet={Object {}}
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  stroke=""
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "color": "#00238a",
                        "height": 20,
                        "width": 20,
                      },
                      Object {
                        "flex": 0,
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                  tintColor={4278199178}
                  vbHeight={24}
                  vbWidth={24}
                >
                  <RNSVGGroup
                    propList={
                      Array [
                        "stroke",
                      ]
                    }
                    stroke={null}
                  >
                    <RNSVGGroup>
                      <RNSVGPath
                        d="M12 9.77778L4.22222 2L2 4.22222L9.77778 12L2 19.7778L4.22222 22L12 14.2222L19.7778 22L22 19.7778L14.2222 12L22 4.22222L19.7778 2L12 9.77778Z"
                        fill={
                          Array [
                            2,
                          ]
                        }
                        propList={
                          Array [
                            "fill",
                            "stroke",
                          ]
                        }
                        stroke={null}
                      />
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
            <View
              aria-orientation="horizontal"
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": "#d4d4d4",
                  "height": 1,
                  "marginBottom": 8,
                  "marginTop": 8,
                  "width": "100%",
                }
              }
              thickness="1"
            />
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "column",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Body
              </Text>
            </View>
            <View
              aria-orientation="horizontal"
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": "#d4d4d4",
                  "height": 1,
                  "marginBottom": 20,
                  "marginTop": 20,
                  "width": "100%",
                }
              }
              thickness="1"
            />
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }
            >
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Footer
              </Text>
            </View>
          </View>
        </View>
      </Modal>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      elevation={0}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": -600,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 600,
          },
        ]
      }
    >
      <View
        dataSet={Object {}}
        style={
          Array [
            Object {
              "backgroundColor": "#eeeeee",
              "flex": 1,
              "justifyContent": "space-between",
              "paddingBottom": 80,
              "paddingTop": 80,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Array [
              Object {},
              Object {
                "dataSet": Object {},
              },
            ]
          }
          data={
            Array [
              Object {
                "icon": <Icon
                  allowFontScaling={false}
                  color="#000000"
                  name="power-off"
                  size={24}
                />,
                "id": "LOGOUT",
                "label": "LOGOUT",
                "onPress": [Function],
              },
            ]
          }
          dataSet={Object {}}
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={Object {}}
          viewabilityConfigCallbackPairs={Array []}
        >
          <View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  dataSet={Object {}}
                  style={
                    Array [
                      Object {
                        "borderBottomWidth": 1,
                        "borderColor": "#7f7f7f",
                        "paddingBottom": 8,
                        "paddingLeft": 16,
                        "paddingRight": 20,
                        "paddingTop": 8,
                      },
                      Array [
                        Object {},
                      ],
                    ]
                  }
                >
                  <View
                    dataSet={Object {}}
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 24,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "width": 12,
                        }
                      }
                    />
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        LOGOUT
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
